src/main/java/com/synopsys/integration/alert/channel/azure/boards/descriptor/AzureBoardsGlobalUIConfig.java

```diff
@@ -0,0 +1,75 @@
+import com.synopsys.integration.alert.common.descriptor.config.field.PasswordConfigField;
+import com.synopsys.integration.alert.common.descriptor.config.field.TextInputConfigField;
+import com.synopsys.integration.alert.common.descriptor.config.field.URLInputConfigField;
+import com.synopsys.integration.alert.common.descriptor.config.field.endpoint.EndpointButtonField;
+import com.synopsys.integration.alert.common.descriptor.config.field.validators.EncryptionValidator;
+import com.synopsys.integration.alert.common.descriptor.config.ui.UIConfig;
+
+@Component
+public class AzureBoardsGlobalUIConfig extends UIConfig {
+    public static final String LABEL_AZURE_BOARDS_URL = "Url";
+    public static final String LABEL_ORGANIZATION_NAME = "Organization Name";
+    public static final String LABEL_CONSUMER_KEY = "Consumer Key";
+    public static final String LABEL_PRIVATE_KEY = "Private Key";
+    public static final String LABEL_OAUTH = "Initialize OAuth";
+    public static final String LABEL_ACCESS_TOKEN = "Access Token";
+
+    public static final String DESCRIPTION_AZURE_BOARDS_URL = "FILL OUT THIS DESCRIPTION";
+    public static final String DESCRIPTION_ORGANIZATION_NAME = "FILL OUT THIS DESCRIPTION";
+    public static final String DESCRIPTION_CONSUMER_KEY = "FILL OUT THIS DESCRIPTION";
+    public static final String DESCRIPTION_PRIVATE_KEY = "FILL OUT THIS DESCRIPTION";
+    public static final String DESCRIPTION_OAUTH = "FILL OUT THIS DESCRIPTION";
+    public static final String DESCRIPTION_ACCESS_TOKEN = "FILL OUT THIS DESCRIPTION";
+
+    public static final String BUTTON_LABEL_OAUTH = "FILL OUT THIS BUTTON LABEL";
+
+    private final EncryptionValidator encryptionValidator;
+
+    @Autowired
+    public AzureBoardsGlobalUIConfig(EncryptionValidator encryptionValidator) {
+        super(AzureBoardsDescriptor.AZURE_BOARDS_LABEL, AzureBoardsDescriptor.AZURE_BOARDS_DESCRIPTION, AzureBoardsDescriptor.AZURE_BOARDS_URL);
+        this.encryptionValidator = encryptionValidator;
+    }
+
+    @Override
+    public List<ConfigField> createFields() {
+        ConfigField azureBoardsUrlField = new URLInputConfigField(AzureBoardsDescriptor.KEY_AZURE_BOARDS_URL, LABEL_AZURE_BOARDS_URL, DESCRIPTION_AZURE_BOARDS_URL).applyRequired(true);
+        ConfigField organizationName = new TextInputConfigField(AzureBoardsDescriptor.KEY_ORGANIZATION_NAME, LABEL_ORGANIZATION_NAME, DESCRIPTION_ORGANIZATION_NAME).applyRequired(true);
+        ConfigField consumerKey = new PasswordConfigField(AzureBoardsDescriptor.KEY_CONSUMER_KEY, LABEL_CONSUMER_KEY, DESCRIPTION_CONSUMER_KEY, encryptionValidator).applyRequired(true);
+        ConfigField privateKey = new PasswordConfigField(AzureBoardsDescriptor.KEY_PRIVATE_KEY, LABEL_PRIVATE_KEY, DESCRIPTION_PRIVATE_KEY, encryptionValidator).applyRequired(true);
+        ConfigField configureOAuth = new EndpointButtonField(AzureBoardsDescriptor.KEY_OAUTH, LABEL_OAUTH, DESCRIPTION_OAUTH, BUTTON_LABEL_OAUTH);
```

Comment:
There isn't a custom endpoint registered in this PR to handle the oauth connection?  Is that intended for a different PR.  Right now this field will not do anything on the server side.  You could at the least register the custom endpoint for this field and log a message to ensure that the server recieves the request when this button is pushed.

Reply:
I agree with Paulo, can you setup a custom endpoint so when the button is pushed it logs a message for now, that way we can easily add the implementation later.